crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -386,54 +401,71 @@ impl SlidingSync {
         Ok(update_summary)
     }
 
-    #[instrument(skip_all, fields(pos))]
-    async fn sync_once(&self) -> Result<UpdateSummary> {
-        let (request, request_config, requested_room_unsubscriptions) = {
-            // Collect requests for lists.
-            let mut requests_lists = BTreeMap::new();
+    async fn generate_sync_request(
+        &self,
+    ) -> Result<(v4::Request, RequestConfig, BTreeSet<OwnedRoomId>)> {
+        let txn_id = TransactionId::new();
```

Comment:
```suggestion
    async fn generate_sync_request(
        &self,
        txn_id: TransactionId,
    ) -> Result<(v4::Request, RequestConfig, BTreeSet<OwnedRoomId>)> {
```

I bet it's better from a design point of view to be able to pass the `TransactionId` as an argument. At least for the sake of testability of this method.

Reply:
Good idea, it's indeed nicer for testing :+1: